mender-update/deployments.cpp

```diff
@@ -0,0 +1,207 @@
+	v2_req->SetHeader("Accept", "application/json");
+	v2_req->SetBodyGenerator(payload_gen);
+
+	string v1_args = "artifact_name=" + http::URLEncode(artifact_name)
+					 + "&device_type=" + http::URLEncode(device_type);
+	auto v1_req = make_shared<http::OutgoingRequest>();
+	v1_req->SetAddress(path::Join(server_url, v1_uri) + "?" + v1_args);
+	v1_req->SetMethod(http::Method::GET);
+	v1_req->SetHeader("Accept", "application/json");
+
+	auto received_body = make_shared<vector<uint8_t>>();
+	auto handle_data = [received_body, api_handler](unsigned status) {
+		if (status == http::StatusOK) {
+			auto ex_j = json::Load(common::StringFromByteVector(*(received_body.get())));
+			if (ex_j) {
+				APIResponse response {optional::optional<json::Json> {ex_j.value()}};
+				api_handler(response);
+			} else {
+				api_handler(expected::unexpected(ex_j.error()));
+			}
+		} else if (status == http::StatusNoContent) {
+			api_handler(APIResponse {optional::nullopt});
+		}
+	};
+
+	http::ResponseHandler header_handler =
+		[received_body, api_handler](http::ExpectedIncomingResponsePtr exp_resp) {
+			if (!exp_resp) {
+				log::Error("Request to check new deployments failed: " + exp_resp.error().message);
+				APIResponse response = expected::unexpected(exp_resp.error());
+				api_handler(response);
+			}
+
+			auto resp = exp_resp.value();
+			received_body->clear();
+			auto body_writer = make_shared<io::ByteWriter>(received_body);
+			body_writer->SetUnlimited(true);
+			resp->SetBodyWriter(body_writer);
+		};
+
+	http::ResponseHandler v1_body_handler =
+		[received_body, api_handler, handle_data](http::ExpectedIncomingResponsePtr exp_resp) {
+			if (!exp_resp) {
+				log::Error("Request to check new deployments failed: " + exp_resp.error().message);
+				APIResponse response = expected::unexpected(exp_resp.error());
+				api_handler(response);
+			}
+			auto resp = exp_resp.value();
+			auto status = resp->GetStatusCode();
+			if ((status == http::StatusOK) || (status == http::StatusNoContent)) {
+				handle_data(status);
+			} else {
+				api_handler(expected::unexpected(MakeError(
+					BadResponseError,
+					"Got unexpected response [" + to_string(status)
+						+ "]: " + resp->GetStatusMessage())));
+			}
+		};
+
+	auto run_v1_fallback = [v1_req, header_handler, v1_body_handler, &client]() {
+		client.AsyncCall(v1_req, header_handler, v1_body_handler);
+	};
+
+	http::ResponseHandler v2_body_handler = [run_v1_fallback, api_handler, handle_data, &loop](
+												http::ExpectedIncomingResponsePtr exp_resp) {
+		if (!exp_resp) {
+			log::Error("Request to check new deployments failed: " + exp_resp.error().message);
+			APIResponse response = expected::unexpected(exp_resp.error());
+			api_handler(response);
+		}
+		auto resp = exp_resp.value();
+		auto status = resp->GetStatusCode();
+		if ((status == http::StatusOK) || (status == http::StatusNoContent)) {
+			handle_data(status);
+		} else if (status == http::StatusNotFound) {
+			log::Warning(
```

Comment:
I think `Info` is enough here. This will happen on all requests to an Open Source server.